modules/javafx.graphics/src/main/java/javafx/scene/robot/Robot.java

```diff
@@ -0,0 +1,321 @@
+     *
+     * @implSpec This is a convenience method that is equivalent to calling
+     * {@link #keyPress(KeyCode)} followed by {@link #keyRelease(KeyCode)}.
+     * @param keyCode the key to type
+     */
+    public void keyType(KeyCode keyCode) {
+        keyPress(keyCode);
+        keyRelease(keyCode);
+    }
+
+    /**
+     * Returns the current mouse x-position in screen coordinates.
+     *
+     * @return the current mouse x-position
+     */
+    public double getMouseX() {
+        return peer.getMouseX();
+    }
+
+    /**
+     * Returns the current mouse y-position in screen coordinates.
+     *
+     * @return the current mouse y-position
+     */
+    public double getMouseY() {
+        return peer.getMouseY();
+    }
+
+    /**
+     * Returns the current mouse (x, y) screen coordinates as a {@link Point2D}.
+     *
+     * @return the current mouse (x,y) screen coordinates
+     */
+    public Point2D getMousePosition() {
+        return new Point2D(getMouseX(), getMouseY());
+    }
+
+    /**
+     * Moves the mouse to the specified (x,y) screen coordinates.
+     *
+     * @param x screen coordinate x to move the mouse to
+     * @param y screen coordinate y to move the mouse to
+     */
+    public void mouseMove(double x, double y) {
+        peer.mouseMove(x, y);
+    }
+
+    /**
+     * Moves the mouse to the (x,y) screen coordinates specified by the
+     * given {@code location}.
+     *
+     * @param location the (x,y) coordinates to move the mouse to
+     */
+    public final void mouseMove(Point2D location) {
+        Objects.requireNonNull(location);
+        mouseMove(location.getX(), location.getY());
+    }
+
+    /**
+     * Presses the specified {@link MouseButton}s.
+     *
+     * @param buttons the mouse buttons to press
+     */
+    public void mousePress(MouseButton... buttons) {
+        peer.mousePress(buttons);
+    }
+
+    /**
+     * Releases the specified {@link MouseButton}s.
+     *
+     * @param buttons the mouse buttons to release
+     */
+    public void mouseRelease(MouseButton... buttons) {
+        peer.mouseRelease(buttons);
+    }
+
+    /**
+     * Clicks the specified {@link MouseButton}s.
+     *
+     * @implSpec This is a convenience method that is equivalent to calling
+     * {@link #mousePress(MouseButton...)} followed by {@link #mouseRelease(MouseButton...)}.
+     * @param buttons the mouse buttons to click
+     */
+    public void mouseClick(MouseButton... buttons) {
+        mousePress(buttons);
+        mouseRelease(buttons);
+    }
+
+    /**
+     * Scrolls the mouse wheel by the specified amount. Positive {@code wheelAmt}
+     * scrolls up whereas negative scrolls down.
+     *
+     * @param wheelAmt the (signed) amount to scroll the wheel
+     */
+    public void mouseWheel(int wheelAmt) {
+        peer.mouseWheel(wheelAmt);
+    }
+
+    /**
+     * Returns the {@link Color} of the pixel at the specified screen coordinates of the
```

Comment:
> of the primary screen

Maybe "relative to the primary screen"? It should be possible to get the pixel color of a location on a secondary screen.